evaluate.py

```diff
@@ -52,6 +51,43 @@
     help='Batch size used during evaluation, adapt based on your memory usage.')
 
 
+def average_precision_score(y_true, y_score):
+    """ Compute average precision (AP) from prediction scores.
+
+    This is a replacement for the scikit-learn version which, while likely more
+    correct does not follow the same protocol as used in the default Market-1501
+    evaluation that first introduced this score to the person ReID field.
+
+    Args:
+        y_true (array): The binary labels for all data points.
+        y_score (array): The predicted scores for each samples for all data
+            points.
+
+    Raises:
+        ValueError if the length of the labels and scores do not match.
+
+    Returns:
+        A float representing the average precision given the predictions.
+    """
+
+    if len(y_true) != len(y_score):
+        raise ValueError('The length of the labels and predictions must match '
+                         'got lengths y_true:{} and y_score:{}'.format(
+                            len(y_true), len(y_score)))
+
+    y_true_sorted = y_true[np.argsort(-y_score, kind='mergesort')]
```

Comment:
Maybe add comment that it's set to `mergesort` because we want stable sorting? Otherwise, who knows, a future commit might remove it thinking it's only for speed or something.

Reply:
Done!